### ISSUE TITLE
boss keeps resetting HP on cron [$15 awarded]

### ISSUE THREAD

**Issue by lettergram** [HabitRPG/habitica#2657]
The Fiery Gryphon keeps reseting its health either at the days roll over, or when a new player logs in... In either case it's pretty ridiculous.

![screen shot 2014-01-30 at 11 02 21 am](https://f.cloud.github.com/assets/1498748/2042380/6e0cefd8-89d0-11e3-9cf2-6de1949a1b54.png)
##

**Comment by lettergram** [comment]
Update it continues to happen, it seems to happen when ever a new player does damage...

![screen shot 2014-01-31 at 8 23 29 am](https://f.cloud.github.com/assets/1498748/2051132/8981319e-8a83-11e3-869e-2ad0a003d078.png)

**Comment by samjross** [comment]
I'm in the same group as Lettergram, just here to back up what he's saying. This is a huge problem -- I don't think the beast is even beatable because of this bug. We've already done hundreds of damage points, and it's not dropped below 240 hp.

**Comment by lettergram** [comment]
Seems to allow us to bring down it's health, I don't know what changed, but the bug seems to not be an issue any more.

**Comment by colegleason** [comment]
I've looked through the relevant code and I don't see where the issue might
be. I imagine there is something specific to your party that I'm missing.
Let me know if it happens again.

**Comment by colegleason** [comment]
@lettergram mentioned to me that his party has people that accepted, rejected, and ignored the quest accept invite and it seemed to glitch until a specific non-participant logged back on. Not sure what the implications of that are.

We should also consider cases where a party member might be in the Inn. Not sure.

**Comment by phillipthelen** [comment]
My Party also experiences this issue. Everybody accepted the quest and one person is in the inn.

Hope this issue can be resolved soon, as an endless bossfight could be rather hard :/

**Comment by Leonadial** [comment]
My party started the Fiery Gryphon quest 2 days ago and had him down to less than 100 health, but today he's back to full strength 300/300. Any fix for this? What info do you need to help troubleshoot?

**Comment by clusterpainter** [comment]
We also had a problem with the Fiery Gryphon's HP resetting. Any way to resolve this?

**Comment by Earthstar** [comment]
My party just had Vice's Shade health regen, so it's an ongoing problem.

**Comment by KatieSt27** [comment]
My party just had the Fiery Gryphon's HP reset after getting down to less than half.  We've beaten the Gryphon 3x before, although this time we have 2 new members.  But we've been going for about 3 days with no issues until just now.

UserID: aa87a105-5770-40a0-aa8d-cdf74973b912

**Comment by DimTwin** [comment]
Group ID: 15f5b304-8426-49e6-82e0-444455a74628
My ID: 91ab86a8-6fb0-4be4-8eeb-7f6b96aef698

Our entire party had the Fiery Gryphon's HP reset about 12h ago (around 1:00 am GMT +3). Not sure about the rest, but for me it happened this way: I switched to the "Party" page and there was no active quests there, although we had the Gryphon dow to 200hp more or less. So I used "Sync" button and the quest window poped up asking me to accept this quest.. again. Well I did that and the HP got reset right there.
Also not long before that one of our members went to the Inn for a vacation (maybe it's related somehow to the bug)

Hope the info helps!!

**Comment by DimTwin** [comment]
And it happened again today. same scenario with accepting the quest all over again..

**Comment by zoebeagle** [comment]
Our party also had this happen, but with Vice today.  This is the first time our party has experienced this bug, and it's also the first time (I believe) all members had not accepted the quest (one party member did not accept).  I know for sure that said party member HAS logged in today.  If more information is needed, let me know :)

**Comment by AineMedias** [comment]
Mister Creeper getting the same thing as @zoebeagle 

A participant of our quest did attack Vice for 2.0 damage. Another participant posted something about his Instant Death, and I replied (thus activating this bug, Vice regained about 110 HP). Don't know any other things other than this!

**Comment by thisisTQ** [comment]
Hi there, 
Our party experienced the same issue yesterday. We're 4 people, all of whom accepted the quest. The boss healed overnight, the last action we took before health was restored was giving a "+1" on an attack and making a comment in the chat window.
It'd be really great if this issue was fixed, neverending quests are not really motivating...

**Comment by AineMedias** [comment]
My issue has been fixed! Don't know how, though.

**Comment by thisisTQ** [comment]
No way - just now it happened again! The boss is back to full 300 HP! This is SO demotivating for our whole group!
It happened after one party member checked back in on the HabitRPG website. As I clicked on her avatar I saw that she had leveled up. I hit the "sync"-button, top right corner, but her current level didn't sync. So I hit the "refresh"-button of my browser (Chrome).
And then I checked back on the boss that was back on 300 HP... Please, do fix this issue, it is draining. :-/

**Comment by thisisTQ** [comment]
![image](https://f.cloud.github.com/assets/7036945/2493278/e67c8fd6-b28a-11e3-92e2-35df8e665659.png)

**Comment by deilann** [comment]
I can assure you that we would _love_ to fix this issue-- the problem is,
we still have no idea what's causing it. I really understand how
frustrating it must be and I'm really, really sorry. But again-- we really
have no idea why this is happening. (Unless someone else has magically
discovered the problem and not shared with the group.)

**Comment by AineMedias** [comment]
@deilann Like I don't know how my bug's fixed!

**Comment by deilann** [comment]
@BuggyCreeper I'm really glad to hear that. But what we don't know is what's triggering the problem. It would be nice to know a way to fix the bug if it happens to folks, but the best solution is for it never to happen. I was more trying to explain to @LaChroes why this hasn't been fixed yet.

**Comment by thisisTQ** [comment]
Thank you very much for your comment, @deilann, I really appreciate the effort. If there's anything I can provide you with to help you to solve the issue, please let me know.
I'll keep an eye on the patterns, perhaps I'm lucky.

**Comment by deilann** [comment]
Thank you @LaChroes! Hopefully soon we'll find a common denominator!

**Comment by DimTwin** [comment]
My `wild guess` is that bosses get bugged when a participating member does smth irregular: dies, rests in inn, leaves his party.. or perhaps it has smth to do with calculations of damage that is to be done to a boss by this member. It may just be impossible for the system to calculate the daily cron and it resets the hp..

umm makes sence to me. what do you guys think?

**Comment by thisisTQ** [comment]
@DimTwin: There were no party member deaths in our party. No rests. Noone left or joined the party. It just... happened.
As far as I was able to trace the issue, it happened when one member of our party refreshed for the day. She did that during the day (we got CET here), not in the morning. That's about the only pattern I could find so far.

**Comment by AineMedias** [comment]
EDIT: I noticed my bug was caused after I sended a message with @ in it. This bug didn't occur this time, though.

**Comment by DimTwin** [comment]
@LaChroes I've noticed the connection with the refresh too. Thats why I related it to the possible damage calculation bug.. And I also have noticed other issues where people get hit for 60+ damage by bosses. So there's definitely smth very wrong with the system.

**Comment by deilann** [comment]
You're only going to notice after a refresh because that's the only way
you'll see an update to the boss.

If people are getting hit for 60+ damage that's likely not an issue with
the system; it's an issue with their dailies not being saved or them not
doing their dailies. 60+ damage is totally doable. Ask Seneca.

**Comment by DimTwin** [comment]
@deilann well I'm not insisting on the idea) just trying to understand what's going on. thx for the explanation

**Comment by deilann** [comment]
Sorry if I seem gruff-- I'm just kind of scatter-brained today and doing
twenty things at once. Whenever I'm short, it's not me trying to refute or
be mean. It's just the way I talk. :)

**Comment by 1seventy** [comment]
Second day of working on the Ghost Stag, and he just jumped back up to 1200 HP- it seemed to happen when party member Earl Freedom logged in. (Screenshot: http://gyazo.com/39911492a5edd7b2996303374ba595ea ) We've done plenty of boss quests just fine before- 5 gryphons, 2 trapper santas, find the cub, 1 hedgehog. 8 of us on this quest, as far as I can tell, no one died or checked into the inn.

**Comment by MANDRO1D** [comment]
Could it be a bug with a party member (or party founder) setting Custom Day Start? I noticed in settings that there are other issues with this feature https://github.com/HabitRPG/habitrpg/issues/1057

**Comment by deilann** [comment]
#1057 may or may not be related to CDS. That was the initial thought, but
some people have experienced it without CDS.

**Comment by ironrhino12** [comment]
When I reset my Cron my issues went away within 24hrs. My party didn't have problems, only me. (also I'm party leader)

**Comment by MANDRO1D** [comment]
@ironrhino12 That's great to know, I'm going to test changing mine to default. How long has it worked without a problem?

**Comment by deilann** [comment]
User @Lyf in the tavern says:

> @Ryan v2.2.6 q.q ok... I don't feel like registering for GritHub, but we're a party of 6 who talk often... None of us have died, I think, someone has a non-default cron, none of us are resting, and all of us accepted the quest

**Comment by thisisTQ** [comment]
So, got the one boss down. But now we got the same issue on the Hedgehog... One thing I noticed is: The bug seems to be related to those members of our party that show up rather infrequently. The last time I was able to trace the bug it happened when one member didn't check back for more than 24 hours.
Could be coincidence, could be a pattern...

**Comment by ironrhino12** [comment]
@MANDRO1D it worked exactly 15 days, then I got it again (for the 3rd time). If I rest my cron, then log in the next morning, everything is normal... for about a second, if i touch anything I return to my parallel universe. Also when my party member logs on I can see it too, then it vanishes. 

I'm pretty sure my bug is different than other's that are here. My other party member has no issues, only me.

**Comment by MANDRO1D** [comment]
I reset my Cron to 0 and I'm still getting the error. I don't think it has to do with that.

**Comment by AineMedias** [comment]
And it's back, now with the Ghost Stag. It happened when user shelad logged in. Don't know any other things!

Edit: It's targets are alternating every cron! Strange!

**Comment by AineMedias** [comment]
It happened over and over again! Come on!

**Comment by deilann** [comment]
@BuggyCreeper Like I've said before, we still don't know why this is happening to some people and some parties. Please also remember that all of the people who work on Habit's code have other jobs and responsibilities. We have to make enough money to pay our bills before we can work on Habit stuff, else we won't be working on Habit stuff at all. If you want to give incentives to developers spending their spare time on this issue, you can add a bounty to it.

I know this is frustrating, but please be patient. The only way we could definitely "fix" this bug with the info we have is to get rid of quests entirely -- despite them working for most users perfectly fine. The developers do not have infinite time nor do they have a secret solution they're just not sharing with the class.

**Comment by AineMedias** [comment]
@deilann So..... this means stopping the quest for a very long time because of a bug. How do I add bounties by the way?

**Comment by deilann** [comment]
No, my point is that I don't know of any way in which we could definitely
"fix" this bug without just getting rid of the quest system entirely. Not
stopping the quest for a long time -- but actually just getting rid of the
system for everyone.

If you go to the top, you'll see a link about how you can add a bounty
through bounty source.

Although I'm curious -- what happens if you leave and reform your party?
(Abort the quest FIRST so the scroll isn't lost, if you try that.)

**Comment by DimTwin** [comment]
@deilann sounded almost like "its your problem guys, not the developrs'. so go ahead and place a bounty so that they will be interested in fixing the bug." Almost..

rough day again? ;)

PS no offence meant.

**Comment by deilann** [comment]
Well, think about it DimTwin -- most of the developers don't get paid.
They're doing volunteer work. This is an Open Source project. If they're
not having this problem, what incentive do they have to fix it besides
doing it out of the kindness of their hearts? Never mind an incredibly
complicated hair ripping out bug, that would _not_ be fun to fix.

Tyler is busy working on super critical bugs that are breaking core game
functionality and is taking a well earned vacation at the moment.

But like... yes, economic incentives do help the developers. We all have to
make ends meet, and working on Habit is time spent not working on projects
that could pay us.

**Comment by deilann** [comment]
Huh, for some reason Bountysource didn't put a bounty link on this one. Not sure why.

**Comment by DimTwin** [comment]
@deilann I'm not complaining about the work of developers. Neither am I defending someone here.

**Comment by deilann** [comment]
I'm just trying to explain why it's extremely frustrating as a developer to
have someone complain about things not being fixed fast enough, and why
bounties are a great way to provide extra incentive to get what you want
fixed, fixed. :)

**Comment by Leonadial** [comment]
When the issue is reported by a given party, are you able to go into the system and confirm that the boss battle was started in x date, y amount of damage was already done and now suddenly the bosses health is full again? Maybe as a workaround until you find the root cause, would it be possible for you guys to manually set the boss health back to what it was before the reset?

**Comment by DimTwin** [comment]
@deilann yeah especially when you can place one ;)

btw getting frustrated every time someone complains is definitely not the best reaction ;) let them. and do whatever work you can.

**Comment by deilann** [comment]
DimTwin, you're acting as if I can simply control my reactions. When people
act entitled to my time, they are being rude. I understand being
frustrated, but the entitlement is not okay. Now, the question is whether
or not posting a reaction is the best idea. And in that, I disagree. I
think that people need to know that they're being incredibly disrespectful.
Otherwise, they won't cease in their actions -- and will likely get even
more frustrated when someone doesn't respond, possibly escalating their
response.

@Leonadial I don't have database access. Only a select few do, and they're
quite busy. There's a lot of privacy concerns with that.

Workarounds like that are tough, even for folks with database access. They
take a lot more time than you might think. Manually resetting stuff can
quickly become a full time data entry job. Lemme put it this way: Tyler's
so busy that I took over manual entering of contributor levels (database
access not needed for that) because it was taking up too much time. I also
manually reset gems doing support, and that takes quite a bit of time.

The data entry I do is probably 5+ hours a week at the moment. Which
doesn't sound like a lot, but you have to remember that's simply me
refunding gems and awarding contributor levels.

**Comment by deilann** [comment]
Also, I'm looking into why Bountysource didn't edit this post.

**Comment by deilann** [comment]
Here's the bounty source link for this ticket:
https://www.bountysource.com/issues/1396584-boss-keeps-resetting-hp-on-cron

**Comment by deilann** [comment]
And putting a bounty on it fixed it.

**Comment by DimTwin** [comment]
@deilann i didn't say that control is easy. and i totally understand you. felt this way too you know. but if you cant control your reactions then who can?

haha adding a new habit 'Get influenced by others'

I wish all this coding stuff was alot easier so you guys could enjoy your project more than tire of it!

**Comment by deilann** [comment]
I don't think people really control whether or not they're offended; they
simply dampen their reaction to it. I also don't think that would be a
constructive habit -- after all, offense is there for a reason. If someone
is being rude, it's not only natural to be offended it's an incredibly
useful emotion. There's no way to be a social, contributing member of
society without having your emotions influenced by others, whether it's
positive or negative. What matters is what you do about it.

And yes, if this bug were an easy one, I'd be all over it. I think it's out
of my scope though, sigh.

**Comment by DimTwin** [comment]
@deilann Well its a +/- habit for me. Exactly because influence can be negative or positive.

And I think it will be positive to stop the offtopic ;) Guess its not the right place to discuss these things.

Thanks for your opinion on the matter. Cant say I totally agree. But i see no point in arguing.

**Comment by deilann** [comment]
I think we can agree to disagree. :)

And can definitely agree that it's good to be respectful of everyone. It's
cool to be frustrated, but please don't taking out your frustration on the
developers who are volunteering their time to make the site better.

Rock on, folks.

**Comment by ironrhino12** [comment]
@deilann is there kind of data collection we could be doing to help this bug get solved? i.e. browser type, computer, history of use, plug ins, ect.

I know nothing about programing, but wanted to offer assistance. Thanks for taking point on this!

**Comment by deilann** [comment]
This is happening server side, afaik. So your browser, etc., should not
affect it. The other issue I've been seeing with collecting data is people
may not be 100% sure when it happens. Frex, I don't think that the last
thing that happened was someone sent a message in party chat -- I think
sending the message updated the party info, revealing that the quest stuff
was reset.

**Comment by thisisTQ** [comment]
Hi there,
our party started the easter egg hunt quest. And the quest restarts over and over again.
I noticed a pattern, though: The bug happens every time one member of our party is offline (doesn't sign on) for more than 24 hours. If the party member logs in after that, the quest seems to reset.
This has already happened two times over the past week.
I've not much insight into the technical stuff here, but perhaps the server gets "confused" if a player logs back in after several days of absence? Perhaps there are multiple commands getting in the way of each other?

**Comment by deilann** [comment]
That is really interesting. And it could actually, I think, explain all the
cases here.

Can you do me a favor, LaChroes and ask all party members to sign in every
day, no matter what? See if that alleviates some of the pain? There's
obviously no control factor for this "study," but it could supply some info.

**Comment by thisisTQ** [comment]
Hi @deilann,
I'll try to convince them to take part in this little experiment. ;)

**Comment by deilann** [comment]
Thanks! It should also, yanno, keep them from dying regularly and help them
actually participate in teh quest. :P

**Comment by ironrhino12** [comment]
@deilann That makes sense.

Do you think it would help if I disband my party, since thats where my issues are coming from?

**Comment by deilann** [comment]
It might!

**Comment by ghost** [comment]
Same problem here with last Vice quest

**Comment by LlamaHobbit** [comment]
Ah, the dread bug has hit our party at last. We are doing the Spirit of Spring quest for the second time and have completed all other quests except the Hedgebeast with no problems. The only thing different this time is one party member was checked into the inn and we knew she wouldn't be back for another several days. So we force started without her. Now the boss keeps regaining health. We have done a total of 289.9 dmg, and the stag is at 1182 HP. By calculating backwards , it seems to have reset at the party leader's cron. I am not the leader, but I did begin the quest since I had the scroll. Hope this info may be useful to anyone working on this problem.

**Comment by deilann** [comment]
This appears to be fixed with https://github.com/HabitRPG/habitrpg/commit/9a4e8d38b46d02475ca97121b5d77a777f8d1f92 which was based off https://github.com/HabitRPG/habitrpg/pull/3451

If anyone continues to have this issue, please post here.
